src/components/Filter/Filter.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable no-console */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable react/button-has-type */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable react/jsx-filename-extension */
+/* eslint-disable react/prop-types */
+
+import React, { useReducer } from 'react';
+import useFilter from '../hooks/useFilter';
+
+const initialState = {
+  status: false,
+  gender: false,
+  species: false,
+  currentStatus: '',
+  currentGender: '',
+  currentSpecies: '',
+};
+
+const ACTIONS = {
+  STATUS: 'status',
+  GENDER: 'gender',
+  SPECIES: 'species',
+  CURRENT_STATUS: 'current-status',
+  CURRENT_GENDER: 'current-gender',
+  CURRENT_SPECIES: 'current-species',
+};
+
+function reducer(state, action) {
+  switch (action.type) {
+    case ACTIONS.STATUS: {
+      return {
+        ...state,
+        status: action.payload,
+      };
+    }
+    case ACTIONS.GENDER: {
+      return {
+        ...state,
+        gender: action.payload,
+      };
+    }
+    case ACTIONS.SPECIES: {
+      return {
+        ...state,
+        species: action.payload,
+      };
+    }
+    case ACTIONS.CURRENT_STATUS: {
+      return {
+        ...state,
+        currentStatus: action.payload,
+      };
+    }
+    case ACTIONS.CURRENT_GENDER: {
+      return {
+        ...state,
+        currentGender: action.payload,
+      };
+    }
+    case ACTIONS.CURRENT_SPECIES: {
+      return {
+        ...state,
+        currentSpecies: action.payload,
+      };
+    }
+    default:
+      return state;
+  }
+}
+
+function Filter(props) {
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const {
+    status, gender, species, currentGender, currentSpecies, currentStatus,
+  } = state;
+  const { getErrorValue, getSearchValue, getDataValue } = props;
+
+  const addition = useFilter(currentGender, currentSpecies, currentStatus);
+
+  const handlerStatus = (event) => {
+    event.preventDefault();
+    return status === false ? dispatch({ type: ACTIONS.STATUS, payload: true })
+      : dispatch({ type: ACTIONS.STATUS, payload: false });
+  };
+
+  const handlerSpecies = (event) => {
+    event.preventDefault();
+    return species === false ? dispatch({ type: ACTIONS.SPECIES, payload: true })
+      : dispatch({ type: ACTIONS.SPECIES, payload: false });
+  };
+
+  const handlerGender = (event) => {
+    event.preventDefault();
+    return gender === false ? dispatch({ type: ACTIONS.GENDER, payload: true })
+      : dispatch({ type: ACTIONS.GENDER, payload: false });
+  };
+
+  const handlerCurrentStatus = (event) => {
+    event.preventDefault();
```

Comment:
do we really need preventDefault every time? what is this method doing?

Reply:
This method prevents potential reload of the page after a button click. So, it's better to use it only with buttons

Reply:
fixed